diff --git a/node_modules/@spectrum-web-components/textfield/src/Textfield.dev.js b/node_modules/@spectrum-web-components/textfield/src/Textfield.dev.js
index 6cdfaba..f88f500 100644
--- a/node_modules/@spectrum-web-components/textfield/src/Textfield.dev.js
+++ b/node_modules/@spectrum-web-components/textfield/src/Textfield.dev.js
@@ -54,6 +54,12 @@ export class TextfieldBase extends ManageHelpText(
     this._value = "";
     this.quiet = false;
     this.required = false;
+    this._eventHandlers = {
+      input: this.handleInput.bind(this),
+      change: this.handleChange.bind(this),
+      focus: this.onFocus.bind(this),
+      blur: this.onBlur.bind(this)
+    };
   }
   static get styles() {
     return [textfieldStyles, checkmarkStyles];
@@ -134,6 +140,23 @@ export class TextfieldBase extends ManageHelpText(
   onBlur() {
     this.focused = !this.readonly && false;
   }
+
+  firstUpdated() {
+    super.firstUpdated();
+    this.inputElement.addEventListener("change", this._eventHandlers["change"]);
+    this.inputElement.addEventListener("input", this._eventHandlers["input"]);
+    this.inputElement.addEventListener("focus", this._eventHandlers["focus"]);
+    this.inputElement.addEventListener("blur", this._eventHandlers["blur"]);
+  }
+
+  disconnectedCallback() {
+    this.inputElement.removeEventListener("change", this._eventHandlers["change"]);
+    this.inputElement.removeEventListener("input", this._eventHandlers["input"]);
+    this.inputElement.removeEventListener("focus", this._eventHandlers["focus"]);
+    this.inputElement.removeEventListener("blur", this._eventHandlers["blur"]);
+    super.disconnectedCallback();
+  }
+
   renderStateIcons() {
     if (this.invalid) {
       return html`
@@ -191,6 +214,7 @@ export class TextfieldBase extends ManageHelpText(
     return html`
             <!-- @ts-ignore -->
             <input
+                data-test-id="patched"
                 type=${this.type}
                 aria-describedby=${this.helpTextId}
                 aria-label=${this.label || this.appliedLabel || this.placeholder}
@@ -206,10 +230,6 @@ export class TextfieldBase extends ManageHelpText(
                 pattern=${ifDefined(this.pattern)}
                 placeholder=${this.placeholder}
                 .value=${live(this.displayValue)}
-                @change=${this.handleChange}
-                @input=${this.handleInput}
-                @focus=${this.onFocus}
-                @blur=${this.onBlur}
                 ?disabled=${this.disabled}
                 ?required=${this.required}
                 ?readonly=${this.readonly}
